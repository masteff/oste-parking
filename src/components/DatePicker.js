import React, { useState } from 'react'
import { connect } from 'react-redux'
import moment from 'moment'
import { SingleDatePicker } from 'react-dates'
import { setActualDate } from '../actions/filters';

const DatePicker = (props) => {

    const [date, setDate] = useState(moment())

    const onDateChange = (date) => {
        if (date) {
            setDate(date)
            props.dispatch(setActualDate(date.valueOf()))
        }
    }

    const highlightDays = (day) => {
        return props.filters.freeDates.some((date) => day.isSame(moment(date), 'day'))
    }

    const onFocusChange = ({ focused }) => {

    }

    return (
        <SingleDatePicker
            date={date}
            onDateChange={onDateChange}
            focused={!props.filters.spotOpen}
            onFocusChange={onFocusChange}
            isDayHighlighted={highlightDays}
            keepOpenOnDateSelect={true}
            numberOfMonths={2}
            orientation='horizontal'
            anchorDirection='right'
            id='1'
        />
    )

}

const mapStateToProps = (state) => ({
    filters: state.filters
})

export default connect(mapStateToProps)(DatePicker)