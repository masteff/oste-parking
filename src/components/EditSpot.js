import React from 'react'

const EditSpot = (props) => (
    <div>
        Here I Will edit the spot with id {props.match.params.spot}
    </div>
)

export default EditSpot