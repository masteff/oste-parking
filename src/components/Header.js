import React from 'react'
import { Link } from 'react-router-dom'
import { connect } from 'react-redux'
import { startLogout } from '../actions/auth'


export const Header = ({ signOut }) => (
    <header>
        <div className="header">
            <div className="content_container">
                <div className="header_content">
                    <h1 className="header__title">oste parking</h1>
                    <button className="button button--link" onClick={signOut}>Logout</button>
                </div>
            </div>
        </div>
        <div className="page-header">
        <div className="header__link_button content_container" >
            <Link className="header__link" to='/u1' >
                <div className="u1">
                    U1
                </div>
            </Link>
            <Link className="header__link" to='/u2' >
                <div className="u2">
                    U2
                </div>
            </Link>
        </div>
        </div>
    </header>
)

const mapDispatchToProps = (dispatch) => ({
    signOut: () => dispatch(startLogout())
})

export default connect(undefined, mapDispatchToProps)(Header)
