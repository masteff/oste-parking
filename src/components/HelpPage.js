import React from 'react'

const HelpPage = () => (
    <div>
        Here you will be provided with HÃ„lp
    </div>
)

export default HelpPage