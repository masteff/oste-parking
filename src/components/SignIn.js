import React, {useState} from 'react'
import { startLogin, startLogout, startCreate, startVerify } from '../actions/auth'
import { connect } from 'react-redux'

const SignIn = (props) => {

    const [email, setEmail] = useState('')
    const [name, setName] = useState('')
    const [password, setPassword] = useState('')

    const onEmailChange = (e) => {
        setEmail(e.target.value)
    }
    const onPasswordChange = (e) => {
        setPassword(e.target.value)
    }
    const onNameChange = (e) => {
        setName(e.target.value)
    }
    const onCreate = () => {
        props.create(email, password, name)
    }
    const onSignIn = () => {
        props.signIn(email, password)
    }
    const onVerify = () => {
        props.verify()
    }

        return (
            <div className="box-layout">
                <div className="box-layout__box">
                    <h1 className="box-layout__title">oste parking</h1>
                    <input type='text' placeholder='E-Mail' onChange={onEmailChange} />
                    <input type='password' placeholder='Passwort' onChange={onPasswordChange} />
                    <input type='text' placeholder='Username' onChange={onNameChange} />
                    <button className="button" onClick={onSignIn}>Einloggen</button>
                    <button className="button" onClick={onCreate}>Registrieren</button>
                    {/* <button onClick={onVerify}>Verify</button> */}
                </div>
            </div>
        )
}

const matchDispatchToProps = (dispatch) => ({
    signIn: (email, password) => dispatch(startLogin(email, password)),
    signOut: () => dispatch(startLogout()),
    create: (email, password, name) => dispatch(startCreate(email, password, name)),
    verify: () => dispatch(startVerify)
})

export default connect(undefined, matchDispatchToProps)(SignIn)