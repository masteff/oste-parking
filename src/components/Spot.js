import React from 'react';
import { connect } from 'react-redux';
import {ownSpot, giveSpot, takeSpot} from '../actions/spots'
import getFreeSpots from '../selectors/spots'
import moment from 'moment'


export const Spot = ({number,owner,actualDate,freeOn,setOwner,setFree,setTaken,taken}) => (
    <div 
    style={freeOn.some((date) => date.isSame(actualDate, 'day')) ? 
    (taken.some((date) => date.takenOn.isSame(actualDate, 'day')) ? 
    divStyleTaken : divStyleFree) 
    : divStyle}
        >
        <p style={pStyle}>{number}</p>
        <p style={pStyle}>{owner}</p>
        <button onClick={() => {
                setOwner({owner:'Tiffany', number})
            }}>own</button>
        <button onClick={() => {
                setFree({number, freeOn: actualDate})
            }}>give</button>
        <button onClick={() => {
                setTaken({number, takenOn: actualDate, takenBy: 'Maddin'})
            }}>take</button> 
    </div>
)

const mapDispatchToProps = (dispatch) => {
    return {
        setOwner: (data) => dispatch(ownSpot(data)),
        setFree: (data) => dispatch(giveSpot(data)),
        setTaken: (data) => dispatch(takeSpot(data))
    }
}
const divStyle = {
    display: 'flex',
    flexDirection: 'column',
    justifyContent: 'center',
    alignItems: 'center',
    height: 150,
    width: 150,
    background: 'lightgrey',
    margin: 5
}

const divStyleFree = {
    display: 'flex',
    flexDirection: 'column',
    justifyContent: 'center',
    alignItems: 'center',
    height: 150,
    width: 150,
    background: 'lightgreen',
    margin: 5
}

const divStyleTaken = {
    display: 'flex',
    flexDirection: 'column',
    justifyContent: 'center',
    alignItems: 'center',
    height: 150,
    width: 150,
    background: 'red',
    margin: 5
}

const pStyle = {
    margin: 0
}


export default connect(undefined,mapDispatchToProps)(Spot)