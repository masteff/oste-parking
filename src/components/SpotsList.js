import React from 'react'
import {connect} from 'react-redux'
import Spot from './Spot'


export const SpotsList = (props) => (
    <div style={{display: 'flex'}}>
        {props.spots.map((spot) => {
            if(props.floor === 'u1' && spot.number <= 10 )
                return <Spot {...spot} {...props.filters} key={spot.number}/>
            else if (props.floor === 'u2' && spot.number >=11 )
                return <Spot {...spot} {...props.filters} key={spot.number}/>
        })}
    </div>
)

const mapStateToProps = (state) => {
    return {
        spots: state.spots,
        filters: state.filters
    }
}

export default connect(mapStateToProps)(SpotsList)

