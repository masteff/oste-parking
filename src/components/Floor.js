import React from 'react'
import SpotsList from './SpotsList'


const Floor = (props) => (
        <SpotsList floor={props.match.path.substring(1,3)}/>
)

export default Floor