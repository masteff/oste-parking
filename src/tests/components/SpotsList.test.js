import React from 'react'
import { shallow} from 'enzyme';
import {SpotsList} from '../../components/SpotsList'
import {spots, filters} from '../fixtures'

test('should render SpotsList with spots',() => {
    const wrapper = shallow(<SpotsList floor='u1' spots={spots} filters={filters} />)
    expect(wrapper).toMatchSnapshot()
})

test('should render SpotsList without spots',() => {
    const wrapper = shallow(<SpotsList floor='u2' spots={spots} filters={filters} />)
    expect(wrapper).toMatchSnapshot()
})

