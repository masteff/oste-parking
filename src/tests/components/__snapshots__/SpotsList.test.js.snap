// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render SpotsList with spots 1`] = `
<div
  style={
    Object {
      "display": "flex",
    }
  }
>
  <Connect(Spot)
    actualDate={"1970-01-01T00:00:02.000Z"}
    freeDates={
      Array [
        "1970-01-01T00:00:03.000Z",
        "1970-01-01T00:00:04.000Z",
      ]
    }
    freeOn={
      Array [
        "1970-01-01T00:00:00.300Z",
      ]
    }
    key="1"
    number={1}
    owner="Besitzer1"
    taken={
      Array [
        Object {
          "takenBy": "Egal",
          "takenOn": "1970-01-01T00:00:00.400Z",
        },
      ]
    }
    takenDates={
      Array [
        "1970-01-01T00:00:00.300Z",
        "1970-01-01T00:00:00.400Z",
      ]
    }
  />
  <Connect(Spot)
    actualDate={"1970-01-01T00:00:02.000Z"}
    freeDates={
      Array [
        "1970-01-01T00:00:03.000Z",
        "1970-01-01T00:00:04.000Z",
      ]
    }
    freeOn={
      Array [
        "1970-01-01T00:00:00.300Z",
      ]
    }
    key="2"
    number={2}
    owner="Besitzer2"
    taken={
      Array [
        Object {
          "takenBy": "Egal",
          "takenOn": "1970-01-01T00:00:00.400Z",
        },
      ]
    }
    takenDates={
      Array [
        "1970-01-01T00:00:00.300Z",
        "1970-01-01T00:00:00.400Z",
      ]
    }
  />
  <Connect(Spot)
    actualDate={"1970-01-01T00:00:02.000Z"}
    freeDates={
      Array [
        "1970-01-01T00:00:03.000Z",
        "1970-01-01T00:00:04.000Z",
      ]
    }
    freeOn={
      Array [
        "1970-01-01T00:00:00.300Z",
      ]
    }
    key="3"
    number={3}
    owner="Besitzer3"
    taken={
      Array [
        Object {
          "takenBy": "Egal",
          "takenOn": "1970-01-01T00:00:00.400Z",
        },
      ]
    }
    takenDates={
      Array [
        "1970-01-01T00:00:00.300Z",
        "1970-01-01T00:00:00.400Z",
      ]
    }
  />
</div>
`;

exports[`should render SpotsList without spots 1`] = `
<div
  style={
    Object {
      "display": "flex",
    }
  }
/>
`;
